data_diff/databases/clickhouse.py

```diff
@@ -0,0 +1,153 @@
+from typing import Optional, Type
+
+import clickhouse_driver.dbapi.connection
+
+from .base import ThreadedDatabase, import_helper, ConnectError
+from .base import MD5_HEXDIGITS, CHECKSUM_HEXDIGITS
+from .database_types import (
+    ColType, Decimal, Float, Integer, FractionalType, Native_UUID, TemporalType, Text, Timestamp
+)
+
+
+@import_helper("clickhouse")
+def import_clickhouse():
+    import clickhouse_driver
+    return clickhouse_driver
+
+
+class SingleConnection(clickhouse_driver.dbapi.connection.Connection):
+    def cursor(self, cursor_factory=None):
+        if not len(self.cursors):
+            _ = super().cursor()
+        return self.cursors[0]
+
+
+class Clickhouse(ThreadedDatabase):
+    TYPE_CLASSES = {
+        'Int8': Integer,
+        'Int16': Integer,
+        'Int32': Integer,
+        'Int64': Integer,
+        'Int128': Integer,
+        'Int256': Integer,
+
+        'UInt8': Integer,
+        'UInt16': Integer,
+        'UInt32': Integer,
+        'UInt64': Integer,
+        'UInt128': Integer,
+        'UInt256': Integer,
+
+        'Float32': Float,
+        'Float64': Float,
+
+        'Decimal': Decimal,
+
+        'UUID': Native_UUID,
+
+        'String': Text,
+        'FixedString': Text,
+
+        'DateTime': Timestamp,
+        'DateTime64': Timestamp,
+
+    }
+    ROUNDS_ON_PREC_LOSS = True
```

Comment:
In the tests you wrote "Clickhouse does not round microseconds when inserting in contrast to PostgreSQL and MySQL.", so sounds like `ROUNDS_ON_PREC_LOSS ` should be False.

Reply:
Done.